internal/pkg/convert/graph.go

```diff
@@ -101,19 +101,46 @@ func (graph *GraphLLB) buildBaseImages() {
 		)...,
 	).Root())
 
+	graph.BaseImages[v1alpha2.Debian] = graph.baseImageProcessor(llb.Image(
+		constants.DefaultDebianImage,
+		llb.WithCustomName(graph.Options.CommonPrefix+"base"),
+	).Run(
+		append(graph.commonRunOptions,
+			llb.Shlex("apt update -qq"),
+			llb.WithCustomName(graph.Options.CommonPrefix+"base-aptupdate"),
+		)...,
+	).Run(
+		append(graph.commonRunOptions,
+			// debian image has /bin/sh symlinked to /bin/dash, let's change it to bash
+			llb.Args([]string{"ln", "-svf", "/bin/bash", "/bin/sh"}),
+			llb.WithCustomName(graph.Options.CommonPrefix+"base-symlink"),
+		)...,
+	).Root())
+
 	graph.BaseImages[v1alpha2.Scratch] = graph.baseImageProcessor(llb.Scratch())
 }
 
 func (graph *GraphLLB) buildChecksummer() {
-	graph.Checksummer = llb.Image(
-		constants.DefaultBaseImage,
-		llb.WithCustomName(graph.Options.CommonPrefix+"cksum"),
-	).Run(
-		append(graph.commonRunOptions,
-			llb.Shlex("apk --no-cache --update add coreutils"),
-			llb.WithCustomName(graph.Options.CommonPrefix+"cksum-apkinstall"),
-		)...,
-	).Root()
+	var checksummer llb.State
+	switch graph.Root.Pkg.Variant {
+	case v1alpha2.Alpine:
+		checksummer = llb.Image(
+			constants.DefaultBaseImage,
+			llb.WithCustomName(graph.Options.CommonPrefix+"cksum"),
+		).Run(
+			append(graph.commonRunOptions,
+				llb.Shlex("apk --no-cache --update add coreutils"),
+				llb.WithCustomName(graph.Options.CommonPrefix+"cksum-apkinstall"),
+			)...,
+		).Root()
+	case v1alpha2.Debian:
+		checksummer = llb.Image(
+			constants.DefaultDebianImage,
+			llb.WithCustomName(graph.Options.CommonPrefix+"cksum"),
+		).With()
```

Comment:
`.With` is used since that only retuns a `llb.State` from `llb.Image` without no-op

Reply:
this is a separate step, so I believe it doesn't have to be changed, it's a like a separate tree root which is independent of the variant.